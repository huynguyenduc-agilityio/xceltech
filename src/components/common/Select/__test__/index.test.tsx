import { render } from '@testing-library/react';

import { Select } from '..';

const mockOption = [
  {
    value: 'sick',
    label: 'Sick',
  },
  {
    value: 'exam',
    label: 'Exam',
  },
];

describe('Select component', () => {
  const commonProps = {
    option: mockOption,
    placeholder: 'Select your relief officer ',
    className: 'bg-blue-light',
    onChange: () => {},
  };

  it('should match snapshot', () => {
    const { asFragment } = render(<Select {...commonProps} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
