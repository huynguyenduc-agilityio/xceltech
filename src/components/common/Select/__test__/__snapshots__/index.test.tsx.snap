// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select component should match snapshot 1`] = `
<DocumentFragment>
  <button
    aria-autocomplete="none"
    aria-controls="radix-:r0:"
    aria-expanded="false"
    class="font-productsans flex h-full w-full items-center justify-between rounded-md border border-input px-3 py-2 text-lg ring-offset-background data-[placeholder]:text-muted-foreground focus:outline-none disabled:cursor-not-allowed disabled:opacity-50 [&>span]:line-clamp-1 bg-blue-light"
    data-placeholder=""
    data-state="closed"
    dir="ltr"
    role="combobox"
    type="button"
  >
    <span
      style="pointer-events: none;"
    >
      Select your relief officer 
    </span>
    <svg
      aria-hidden="true"
      class="lucide lucide-chevron-down h-4 w-4 opacity-50"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m6 9 6 6 6-6"
      />
    </svg>
  </button>
</DocumentFragment>
`;
